velox/exec/tests/TableWriteTest.cpp

```diff
@@ -2126,108 +2126,122 @@ TEST_P(AllTableWriterTest, commitStrategies) {
+        for (int i = 0; i < numBatches; i++) {
+          writeToFile(inputFilePaths[i]->getPath(), vectors[i]);
+        }
 
-  auto task = assertQuery(plan, inputFilePaths, "SELECT count(*) FROM tmp");
+        auto outputDirectory = TempDirectoryPath::create();
+        auto plan = createInsertPlan(
+            PlanBuilder().tableScan(rowType),
+            rowType,
+            outputDirectory->getPath(),
+            partitionKeys,
+            bucketProperty_,
+            compressionKind_,
+            getNumWriters(),
+            connector::hive::LocationHandle::TableType::kNew,
+            commitStrategy_);
 
-  std::set<std::string> actualPartitionDirectories =
-      getLeafSubdirectories(outputDirectory->getPath());
-
-  std::set<std::string> expectedPartitionDirectories;
-  const std::vector<std::string> expectedCharsAfterEscape = {
-      "%22",
-      "%23",
-      "%25",
-      "%27",
-      "%2A",
-      "%2F",
-      "%3A",
-      "%3D",
-      "%3F",
-      "%5C",
-      "%7F",
-      "%7B",
-      "%5B",
-      "%5D",
-      "%5E"};
-  for (auto i = 0; i < numPartitions; ++i) {
-    // url encoded
-    auto partitionName = fmt::format(
-        "p0={}/p1=str_{}{}", i, i, expectedCharsAfterEscape.at(i % 15));
-    expectedPartitionDirectories.emplace(
-        fs::path(outputDirectory->getPath()) / partitionName);
-  }
-  EXPECT_EQ(actualPartitionDirectories, expectedPartitionDirectories);
+        auto task =
+            assertQuery(plan, inputFilePaths, "SELECT count(*) FROM tmp");
+
+        std::set<std::string> actualPartitionDirectories =
+            getLeafSubdirectories(outputDirectory->getPath());
+
+        std::set<std::string> expectedPartitionDirectories;
+        const std::vector<std::string> expectedCharsAfterEscape = {
+            "%22",
+            "%23",
+            "%25",
+            "%27",
+            "%2A",
+            "%2F",
+            "%3A",
+            "%3D",
+            "%3F",
+            "%5C",
+            "%7F",
+            "%7B",
+            "%5B",
+            "%5D",
+            "%5E"};
+        for (auto i = 0; i < numPartitions; ++i) {
+          // url encoded
+          std::string partitionName;
+          if (partitionKeys == std::vector<std::string>{"p0", "p1"}) {
+            partitionName = fmt::format(
+                "p0={}/p1=str_{}{}", i, i, expectedCharsAfterEscape.at(i % 15));
+          } else if (partitionKeys == std::vector<std::string>{"p1", "p0"}) {
+            partitionName = fmt::format(
+                "p1=str_{}{}/p0={}", i, expectedCharsAfterEscape.at(i % 15), i);
+          }
+          expectedPartitionDirectories.emplace(
+              fs::path(outputDirectory->getPath()) / partitionName);
+        }
+        EXPECT_EQ(actualPartitionDirectories, expectedPartitionDirectories);
+      };
+  testSpecialPartitionName({"p0", "p1"}, {VARCHAR(), INTEGER()});
+  testSpecialPartitionName({"p1", "p0"}, {INTEGER(), VARCHAR()});
```

Comment:
Can we add a test for the error condition? `partitionKeyOrder does not contain all the partition keys`

Reply:
@majetideepak Yes. Added the error condition test.